mesonbuild/dependencies/base.py

```diff
@@ -1055,21 +1055,26 @@ def _original_module_name(self, module: str) -> str:
         return module
 
     def __init__(self, name: str, environment: Environment, kwargs, language: str = None):
+        # Gather a list of all languages to support
+        self.language_list = []  # type: T.List[str]
         if language is None:
+            compilers = None
             if kwargs.get('native', False):
-                if 'c' in environment.coredata.compilers.build.keys():
-                    language = 'c'
-                elif 'cpp' in environment.coredata.compilers.build.keys():
-                    language = 'cpp'
-                elif 'fortran' in environment.coredata.compilers.build.keys():
-                    language = 'fortran'
+                compilers = environment.coredata.compilers.build
             else:
-                if 'c' in environment.coredata.compilers.host.keys():
-                    language = 'c'
-                elif 'cpp' in environment.coredata.compilers.host.keys():
-                    language = 'cpp'
-                elif 'fortran' in environment.coredata.compilers.host.keys():
-                    language = 'fortran'
+                compilers = environment.coredata.compilers.host
+
+            candidates = ['c', 'cpp', 'fortran', 'objc', 'objcxx']
+            self.language_list += [x for x in candidates if x in compilers.keys()]
```

Comment:
this really should just be:
```python
self.language_list += [x for x in candidates if x in compilers]
```
With is log(n) (it's a hash lookup), as opposed to the `keys()` version with is o(n)